### PR TITLE
Autoloader checker

### PR DESCRIPTION
**Pull request by rullzer**
Lets see if travis correctly fails ;)

### CODE REVIEW

**Comment by rullzer** [comment]
Whee :)

CC: @MorrisJobke @nickvergessen @icewind1991 @LukasReschke

**Comment by nickvergessen** [comment]
:+1:

**Comment by nickvergessen** [comment]
Although I hope that this is not a too big hurdle for contributors :P

**Comment by rullzer** [comment]
Well we tell them what and how. And otherwise we can quickly spot it from the output.

**Comment by LukasReschke** [comment]
Nice! 🚀 👍

File: build/autoloaderchecker.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+if [ -e "composer.phar" ]
+then
+    echo "Composer found: checking for update"
+    php composer.phar self-update
+else
+    echo "Composer not found: fetching"
+    php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
+    php -r "if (hash_file('SHA384', 'composer-setup.php') === 'e115a8dc7871f15d853148a7fbac7da27d6c0030b848d9b3dc09e2a0388afed865e6a3d6b3c0fad45c48e2b5fc1196ae') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;"
+    php composer-setup.php
+    php -r "unlink('composer-setup.php');"
```
Comment (MorrisJobke) [inline build/autoloaderchecker.sh]: `rm composer-setup.php` 🙈 

But you are right - we are PHP people 😆

Comment (rullzer) [inline build/autoloaderchecker.sh]: hehe ah I just copied the stuff from get-composer :wink:

Comment (MorrisJobke) [inline build/autoloaderchecker.sh]: :D

**Pull request merged**